### PR TITLE
feat: undo-redo

### PR DESCRIPTION
**Pull request by DGMC117**
feat: Undo-Redo

### CODE REVIEW

File: Source/Modules/ModuleCommand.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+ModuleCommand::ModuleCommand() : commandListIterator(std::begin(commandList))
+{
+}
+
+ModuleCommand::~ModuleCommand()
+{
+}
+
+bool ModuleCommand::Init()
```
Comment (gerard-queralt) [inline Source/Modules/ModuleCommand.cpp:14]: Can be removed.

File: Source/Modules/ModuleCommand.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+{
+	return true;
+}
+
+update_status ModuleCommand::Update()
+{
+	if (
```
Comment (gerard-queralt) [inline Source/Modules/ModuleCommand.cpp:21]: Bracket should go on the next line.

File: Source/Modules/ModuleCommand.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		App->input->GetKey(SDL_SCANCODE_LCTRL) == KeyState::REPEAT
+		&& App->input->GetKey(SDL_SCANCODE_Z) == KeyState::DOWN)
+	{
+		Undo();
+	}
+	else if (
```
Comment (gerard-queralt) [inline Source/Modules/ModuleCommand.cpp:27]: Bracket should go on the next line.

File: Source/Modules/ModuleCommand.cpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		App->input->GetKey(SDL_SCANCODE_LCTRL) == KeyState::REPEAT
+		&& App->input->GetKey(SDL_SCANCODE_Y) == KeyState::DOWN)
+	{
+		Redo();
+	}
+	return update_status::UPDATE_CONTINUE;
+}
+
+bool ModuleCommand::Cleanup()
```
Comment (gerard-queralt) [inline Source/Modules/ModuleCommand.cpp:36]: Can be removed.

**Review by gerard-queralt — COMMENTED** [review]
Left some minor comments. Thank you for all your work David!

**Review by AuxPort66 — APPROVED** [review]

**Review by Dedado3000 — APPROVED** [review]

**Review by gerard-queralt — APPROVED** [review]

**Pull request merged**